egs2/jkac/tts1/local/prep_segments.py

```diff
@@ -0,0 +1,140 @@
+        description="Prepare segments from text files in yaml format",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument("input_dir", type=str, help="path to J-KAC corpus")
+    parser.add_argument("wav_scp_path", type=str, help="path to output 'wav.scp' file")
+    parser.add_argument("utt2spk_path", type=str, help="path to output 'utt2spk' file")
+    parser.add_argument("text_path", type=str, help="path to output 'text' file")
+    parser.add_argument("segments_path", type=str, help="path to output 'segments' file")
+    parser.add_argument("sample_rate", type=str, help="sampling rate")
+    return parser
+
+
+def list_labels(root_path):
+    txt_dir_path = os.path.join(root_path, "txt")
+    wav_dir_path = os.path.join(root_path, "wav")
+    categories = os.listdir(txt_dir_path)
+    for category in categories:
+        category_txt_path = os.path.join(txt_dir_path, category)
+        category_wav_path = os.path.join(wav_dir_path, category)
+        for label_filename in os.listdir(category_txt_path):
+            if label_filename.endswith(".yaml"):
+                title = label_filename.replace(".yaml", "")
+                label_path = os.path.join(category_txt_path, label_filename)
+                wav_path = os.path.join(category_wav_path, title + ".wav")
+                yield JKACPath(label_path=label_path,
+                               wav_path=wav_path,
+                               category=category,
+                               title=title)
+
+
+def read_label(path):
+    with open(path.label_path, "r") as f:
+        label_dict = yaml.load(f, Loader=yaml.Loader)
+        return parse_label(label_dict, path)
+
+
+def parse_label(book_dict, path):
+    for chapter_id in book_dict.keys():
+        chapter = book_dict[chapter_id]
+        for paragraph_id in chapter.keys():
+            paragraph = chapter[paragraph_id]
+            for style_id in paragraph.keys():
+                style = paragraph[style_id]
+                for sentence_id, sentence in enumerate(style):
+                    yield JKACLabel(path=path,
+                                    chapter_id=chapter_id,
+                                    paragraph_id=paragraph_id,
+                                    style_id=style_id,
+                                    sentence_id=sentence_id + 1,
+                                    sentence=sentence["sent"],
+                                    time_begin=sentence["time"][0],
+                                    time_end=sentence["time"][1])
+
+
+if __name__ == "__main__":
+    args = get_parser().parse_args(sys.argv[1:])
+    sample_rate = None if args.sample_rate == "48000" else args.sample_rate
+
+    with open(args.wav_scp_path, "w") as wav_scp_f:
+        with open(args.utt2spk_path, "w") as utt2spk_f:
+            with open(args.text_path, "w") as text_f:
+                with open(args.segments_path, "w") as segments_f:
```

Comment:
Just FYI: with can concatenate with ","
https://github.com/espnet/espnet/blob/3bbc844fbc3f0a51e208ffc6e57e0f083dd1c582/espnet2/bin/tts_inference.py#L289-L301
This is just a preference.